extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQWorkerTaskLauncher.java

```diff
@@ -460,7 +467,9 @@ private void sleep(final long sleepMillis, final boolean shuttingDown) throws In
       } else {
         // wait on taskIds so we can wake up early if needed.
         synchronized (taskIds) {
-          taskIds.wait(sleepMillis);
+          if (acknowledgedDesiredTaskCount == desiredTaskCount) {
```

Comment:
Nit: Can we add a note here. Something like  :
As desiredTaskCount can change in `launchTasksIfNeeded` which can be called externally from any thread, we need to way to wake up the main loop in case desiredTaskCount is changed.

Reply:
OK, added!